src/neuromaps_prime/transforms/utils.py

```diff
@@ -38,3 +40,62 @@ def get_vertex_count(surface_file: Path) -> int:
     if not isinstance(surface, nib.GiftiImage):
         raise TypeError("Input file is not a GIFTI surface file.")
     return surface.darrays[0].data.shape[0]
+
+
+def validate(
+    graph: NeuromapsGraph,
+    source: str,
+    target: str,
+) -> None:
+    """Validate that source and target spaces exist in the graph."""
+    if source not in graph.nodes(data=False) or target not in graph.nodes(data=False):
+        raise ValueError(
+            f"Either source space '{source}' or "
+            f"target space '{target}' does not exist in the graph."
+            f" Available spaces: {list(graph.nodes(data=False))}"
+        )
```

Comment:
I like the idea here - I think it is better to split this up to source AND target rather than so that we can alert the user to exactly what is missing.